app/src/main/java/at/tugraz/ist/swe/note/MainActivity.java

```diff
@@ -56,30 +54,42 @@ public void onClick(View view) {
     }
 
     @Override
-    protected void onActivityResult(int requestCode, int resultCode, Intent data){
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
         if (requestCode == NOTE_REQUEST_CODE) {
-            if(resultCode == RESULT_OK){
+            if (resultCode == RESULT_OK) {
                 Note note = (Note) data.getSerializableExtra("note");
-                boolean editFlag = (boolean) data.getSerializableExtra("editFlag");
-                if(editFlag)
-                {
-                    try {
-
-                        mNoteStorage.update(note);
-                        recreate();
+                OptionFlag flag = (OptionFlag) data.getSerializableExtra("flag");
 
-                    } catch (NotFoundException e) {
-                        e.printStackTrace();
-                    }
-                }else
+                switch (flag)
                 {
-                    mNoteList.add(note);
-                    mNoteStorage.insert(note);
+                    case SAVE:
+                        mNoteList.add(note);
+                        mNoteStorage.insert(note);
+                        break;
+                    case EDIT:
+                        try {
+
+                            mNoteStorage.update(note);
+                            recreate();
```

Comment:
Why do we need to restart/recreate main activity here? do you do it just to re-read the database?

Reply:
I tried the mCustomNoteAdapter.notifyDataSetChanged() but it doesn't work on edit Note.
For deleting we do not have to call the recreate() - you are right

Reply:
For the edit it is possible to not recreate the activity by calling `mNoteList(int pos, Note newNote)`. What this does is it replaces the note at the given position with the given new note. Then you can call `mCustomNoteAdapter.notifyDataSetChanged();`. The only issue is that it might be hard to get the original note position in the list.